From 76fd220b866da827d0a1bf45ae1f72390e732aee Mon Sep 17 00:00:00 2001
From: Yu Yue <yuyue@mprc.pku.edu.cn>
Date: Tue, 1 Nov 2011 13:48:21 +0800
Subject: [PATCH 055/641] UniCore64: Add struct cpuinfo_op

Fixed error:
file-mmu.c:(.text+0x691f8): undefined reference to `cpuinfo_op'

Signed-off-by: Yu Yue <yuyue@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/Makefile   |    2 +-
 arch/unicore64/kernel/cpu-ucv3.c |   65 ++++++++++++++++++++++++++++++++++++++
 2 files changed, 66 insertions(+), 1 deletion(-)
 create mode 100644 arch/unicore64/kernel/cpu-ucv3.c

diff --git a/arch/unicore64/kernel/Makefile b/arch/unicore64/kernel/Makefile
index eea7b84..bf6b5c6 100644
--- a/arch/unicore64/kernel/Makefile
+++ b/arch/unicore64/kernel/Makefile
@@ -1,4 +1,4 @@
 obj-y			:= ptrace.o setup.o init_task.o process.o
-obj-y			+= time.o traps.o irq.o
+obj-y			+= time.o traps.o irq.o cpu-ucv3.o
 head-y			:= head.o
 extra-y			:= $(head-y) vmlinux.lds
diff --git a/arch/unicore64/kernel/cpu-ucv3.c b/arch/unicore64/kernel/cpu-ucv3.c
new file mode 100644
index 0000000..9681bd0
--- /dev/null
+++ b/arch/unicore64/kernel/cpu-ucv3.c
@@ -0,0 +1,65 @@
+#include <linux/kernel.h>
+#include <linux/seq_file.h>
+
+/**
+ * cpu_ucv3_show() -
+ * @m:
+ * @v:
+ */
+static int cpu_ucv3_show(struct seq_file *m, void *v)
+{
+	/* FIXME */
+	BUG();
+	return 0;
+}
+
+/**
+ * cpu_ucv3_start() -
+ * @m:
+ * @pos:
+ */
+static void *cpu_ucv3_start(struct seq_file *m, loff_t *pos)
+{
+	/* FIXME */
+	BUG();
+}
+
+/**
+ * cpu_ucv3_next() -
+ * @v:
+ * @pos:
+ */
+static void *cpu_ucv3_next(struct seq_file *m, void *v, loff_t *pos)
+{
+	/* FIXME */
+	BUG();
+}
+
+/**
+ * cpu_ucv3_stop() -
+ * @m:
+ * @v:
+ */
+static void cpu_ucv3_stop(struct seq_file *m, void *v)
+{
+	/* FIXME */
+	BUG();
+}
+
+/**
+ * struct cpuinfo_op -
+ * @start:
+ * @next:
+ * @stop:
+ * @show:
+ */
+const struct seq_operations cpuinfo_op = {
+	.start	= cpu_ucv3_start,
+	.next	= cpu_ucv3_next,
+	.stop	= cpu_ucv3_stop,
+	.show	= cpu_ucv3_show
+};
+
+#ifdef CONFIG_CPU_FREQ
+	/* FIXME */
+#endif
-- 
1.7.9.5

